File: src/Domain/Entities/Produits/Produit.cs
using Domain.Common;

namespace Domain.Entities.Produits;

public class Produit : AuditableAndSoftDeletableEntity ,ISanitizable
{
   public  string IdProduit { get;  set; } = default!;
   
   public string Description { get;   set; } = default!;
   
   public decimal Prix { get;   set; } = default!;
   
   public string Categorie { get;   set; } = default!;
   
   public string Format { get;   set; } = default!;
   
   public string Couleur { get;  set; } = default!;
   
   public Byte Quantite { get;   set; } = default!;
   
   public string[] PhotoUrl { get;   set; } = [];
   

   public void SetPhotoUrl(string[] photoUrl)
   {
      PhotoUrl = photoUrl;
   }
   
   public void SanitizeForSaving()
   {
      IdProduit = IdProduit.Trim();
      Description = Description.Trim();
      
   }
}

File: src/Web/vue-app/src/types/entities/produit.ts
import {Couleur} from "@/types/entities/couleur";
import {Categorie} from "@/types/entities/categorie";
import {Format} from "@/types/entities/format";

export class Produit {
    idProduit?: string;
    description?: string;
    prix?: number;
    categorie?: string;
    format?: string;
    couleur?: string;
    quantite?: number;
    photosUrl?: string [];
}